src/main/java/org/sonarsource/sonarlint/ls/SonarLintLanguageServer.java

```diff
@@ -133,17 +144,59 @@
     this.serverNotifications = new ServerNotifications(client, workspaceFoldersManager, telemetry, lsLogOutput);
     this.settingsManager.addListener((WorkspaceSettingsChangeListener) serverNotifications);
     this.settingsManager.addListener((WorkspaceFolderSettingsChangeListener) serverNotifications);
-    this.analysisManager = new AnalysisManager(lsLogOutput, enginesFactory, client, telemetry, workspaceFoldersManager, settingsManager, bindingManager);
+    this.analysisManager = new AnalysisManager(lsLogOutput, enginesFactory, client, telemetry, workspaceFoldersManager, settingsManager, bindingManager, fileTypeClassifier,
+      fileLanguageCache, this::getJavaConfigFromCacheOrFetch);
+    this.workspaceFoldersManager.addListener(analysisManager);
     bindingManager.setAnalysisManager(analysisManager);
     this.settingsManager.addListener(analysisManager);
     this.commandManager = new CommandManager(client, settingsManager, bindingManager, analysisManager, telemetry);
     this.securityHotspotsHandlerServer = new SecurityHotspotsHandlerServer(lsLogOutput, bindingManager, client, telemetry);
     launcher.startListening();
   }
 
-  static SonarLintLanguageServer bySocket(int port, Collection<URL> analyzers) throws IOException {
+  static void bySocket(int port, Collection<URL> analyzers) throws IOException {
     Socket socket = new Socket("localhost", port);
-    return new SonarLintLanguageServer(socket.getInputStream(), socket.getOutputStream(), analyzers);
+    new SonarLintLanguageServer(socket.getInputStream(), socket.getOutputStream(), analyzers);
+  }
+
+  public Optional<SonarLintExtendedLanguageClient.GetJavaConfigResponse> getJavaConfigFromCacheOrFetch(URI fileUri) {
+    Optional<SonarLintExtendedLanguageClient.GetJavaConfigResponse> javaConfigOpt;
+    try {
+      javaConfigOpt = getJavaConfigFromCacheOrFetchAsync(fileUri).get(1, TimeUnit.MINUTES);
+    } catch (InterruptedException e) {
+      Utils.interrupted(e);
+      javaConfigOpt = empty();
+    } catch (Exception e) {
+      LOG.warn("Unable to get Java config", e);
+      javaConfigOpt = empty();
+    }
+    return javaConfigOpt;
+  }
+
+  /**
+   * Try to fetch Java config. In case of any error, cache an empty result to avoid repeted calls.
+   */
+  private CompletableFuture<Optional<SonarLintExtendedLanguageClient.GetJavaConfigResponse>> getJavaConfigFromCacheOrFetchAsync(URI fileUri) {
```

Comment:
I'm not sure why this was moved from `AnalysisManager`, especially since it makes `AnalysisManager#javaConfigPerFileURI` `public`.

Reply:
I created a JavaConfigCache class as discussed